These two functions are deprecated and not required with OpenSSL 1.1. They
are usually present as a nop but the OmniOS openssl is set to API version 1.0
and above.

diff -wpruN '--exclude=*.orig' a~/libvncclient/tls_openssl.c a/libvncclient/tls_openssl.c
--- a~/libvncclient/tls_openssl.c	1970-01-01 00:00:00
+++ a/libvncclient/tls_openssl.c	1970-01-01 00:00:00
@@ -101,8 +101,6 @@ static rfbBool InitLockingCb()
   for (i = 0; i < CRYPTO_num_locks(); i++)
     INIT_MUTEX(mutex_buf[i]);
 
-  CRYPTO_set_locking_callback(locking_function);
-  CRYPTO_set_id_callback(id_function);
   CRYPTO_set_dynlock_create_callback(dyn_create_function);
   CRYPTO_set_dynlock_lock_callback(dyn_lock_function);
   CRYPTO_set_dynlock_destroy_callback(dyn_destroy_function);
